<?xml version="1.0" encoding="utf-16"?>
<!-- Changes:
18-Dec-2017 - Dirk Schiwy, D033833
- Initial version, Dirk Schiwy
 -->
<targetsystem desc="BIO_HANA_CFG - All Checks" id="BIO_HANACFG_ALL" multisql="Yes">
  <configstore name="HDB_PARAMETER">
    <checkitem desc="Listen Interface" id="CFG.HANA.01">
      <compliant>FILE_NAME = 'global.ini' and 
                 SECT = 'communication' and 
                 NAME = 'listeninterface' and 
                 VALUE != '.global'
      </compliant>
      <noncompliant>FILE_NAME = 'global.ini' and 
                 SECT = 'communication' and 
                 NAME = 'listeninterface' and 
                 NOT ( VALUE  != '.global' )
      </noncompliant>
    </checkitem>
    <!-- is being done differently by security compliance , remove for the time being
    <checkitem desc="SSL self signed Certificates" id="CFG.HANA.02">
      <compliant>FILE_NAME = 'global.ini' and 
                 SECT = 'communication' and 
                 NAME = 'sslcreateselfsignedcertificate' and 
                 VALUE = 'true'
      </compliant>
      <noncompliant>FILE_NAME = 'global.ini' and 
                 SECT = 'communication' and 
                 NAME = 'sslcreateselfsignedcertificate' and 
                 NOT ( VALUE  = 'true' )
      </noncompliant>
    </checkitem>
-->
    <checkitem desc="Audit Configuration" id="CFG.HANA.03">
      <compliant>FILE_NAME = 'global.ini' and 
                 SECT = 'auditing configuration' and 
                 NAME = 'default_audit_trail_type' and 
                 VALUE = 'SYSLOGPROTOCOL'
      </compliant>
      <noncompliant>FILE_NAME = 'global.ini' and 
                 SECT = 'auditing configuration' and 
                 NAME = 'default_audit_trail_type' and 
                 NOT ( VALUE = 'SYSLOGPROTOCOL' )
      </noncompliant>
    </checkitem>
    <checkitem desc="Audit Configuration" id="CFG.HANA.04">
      <compliant>FILE_NAME = 'global.ini' and 
                 SECT = 'auditing configuration' and 
                 NAME = 'global_auditing_state' and 
                 VALUE = 'true'
      </compliant>
      <noncompliant>FILE_NAME = 'global.ini' and 
                 SECT = 'auditing configuration' and 
                 NAME = 'global_auditing_state' and 
                 NOT ( VALUE = 'true' )
      </noncompliant>
    </checkitem>
    <checkitem desc="Persistence Configuration" id="CFG.HANA.05">
      <compliant>FILE_NAME = 'global.ini' and 
                 SECT = 'persistence' and 
                 NAME = 'basepath_logbackup' and 
                 VALUE like '/hana/backup/log/%'
      </compliant>
      <noncompliant>FILE_NAME = 'global.ini' and 
                 SECT = 'persistence' and 
                 NAME = 'basepath_logbackup' and 
                 VALUE not like '/hana/backup/log/%'
      </noncompliant>
    </checkitem>
    <checkitem desc="Persistence Configuration" id="CFG.HANA.06">
      <compliant>FILE_NAME = 'global.ini' and 
                 SECT = 'persistence' and 
                 NAME = 'basepath_databackup' and 
                 ( VALUE like '/hana/backup/%/data' OR VALUE like '/hana_backup/%/data' )
      </compliant>
      <noncompliant>FILE_NAME = 'global.ini' and 
                 SECT = 'persistence' and 
                 NAME = 'basepath_databackup' and 
                 NOT ( VALUE like '/hana/backup/%/data' OR VALUE like '/hana_backup/%/data'  )
      </noncompliant>
    </checkitem>
    <checkitem desc="Backup Configuration" id="CFG.HANA.07">
      <compliant>FILE_NAME = 'global.ini' and 
                 SECT = 'backup' and 
                 NAME = 'data_backup_max_chunk_size' and 
                 VALUE = '512000'
      </compliant>
      <noncompliant>FILE_NAME = 'global.ini' and 
                SECT = 'backup' and 
                 NAME = 'data_backup_max_chunk_size' and 
                 NOT VALUE = '512000'
      </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>