<?xml version="1.0" encoding="utf-16"?>
<!-- Changes:
18-Dec-2017 - Dirk Schiwy, D033833
- Initial version, Dirk Schiwy
 -->
<targetsystem desc="BSM_ABAP_SEC - All Checks" id="BSM_ABAP_SEC_ALL" multisql="Yes">
  <!-- Checks for client settings -->
  <configstore name="CLIENTS" name_extended="" sci_id="">
    <checkitem desc="Delete unused clients (001)" id="1.10.01a" operator="not_exist:MANDT">
      <compliant>MANDT = '001'</compliant>
      <noncompliant>MANDT='001'</noncompliant>
    </checkitem>
    <checkitem desc="Delete unused clients (066)" id="1.10.01b" operator="not_exist:MANDT">
      <compliant>MANDT='066'</compliant>
      <noncompliant>MANDT='066'</noncompliant>
    </checkitem>
    <checkitem desc="System and Client Change options (client 000)" id="2.90.02b" operator="">
      <compliant>MANDT = '000' and CCCORACTIV = '2' and CCNOCLIIND = '3' and CCCOPYLOCK = 'X' and CCIMAILDIS like '' </compliant>
      <noncompliant>MANDT = '000' and not ( CCCORACTIV = '2' and CCNOCLIIND = '3' and CCCOPYLOCK = 'X' and CCIMAILDIS like '' ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for profile parametes -->
  <configstore name="ABAP_INSTANCE_PAHI" name_extended="" sci_id="">
    <checkitem desc="Secure SAP* (login/no_automatic_user_sapstar)" id="1.10.02a">
      <compliant>NAME = 'login/no_automatic_user_sapstar' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'login/no_automatic_user_sapstar' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_lng)" id="1.20.01a">
      <compliant>NAME = 'login/min_password_lng' and VALUE &gt;= '8' </compliant>
      <noncompliant>NAME = 'login/min_password_lng' and not ( VALUE &gt;= '8' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_letters)" id="1.20.01b">
      <compliant>NAME = 'login/min_password_letters' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_letters' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_digits)" id="1.20.01c">
      <compliant>NAME = 'login/min_password_digits' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_digits' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_lowercase)" id="1.20.01d">
      <compliant>NAME = 'login/min_password_lowercase' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_lowercase' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_uppercase)" id="1.20.01e">
      <compliant>NAME = 'login/min_password_uppercase' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_uppercase' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_expiration_time)" id="1.20.01f">
      <compliant>NAME = 'login/password_expiration_time' and replace(VALUE, ' (not 0)' , '9999') between 1 and 182</compliant>
      <noncompliant>NAME = 'login/password_expiration_time' and not ( replace(VALUE, ' (not 0)' , '9999') between 1 and 182 ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_history_size)" id="1.20.01g">
      <compliant>NAME = 'login/password_history_size' and VALUE &gt;= '5' </compliant>
      <noncompliant>NAME = 'login/password_history_size' and not ( VALUE &gt;= '5' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_change_waittime)" id="1.20.01h">
      <compliant>NAME = 'login/password_change_waittime' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/password_change_waittime' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_compliance_to_current_policy)" id="1.20.01i">
      <compliant>NAME = 'login/password_compliance_to_current_policy' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'login/password_compliance_to_current_policy' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (login/password_downwards_compatibility)" id="1.20.02a">
      <compliant>NAME = 'login/password_downwards_compatibility' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'login/password_downwards_compatibility' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/enable)" id="1.50.02a">
      <compliant>NAME = 'snc/enable' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'snc/enable' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/data_protection/max)" id="1.50.02b">
      <compliant>NAME = 'snc/data_protection/max' and VALUE = '3' </compliant>
      <noncompliant>NAME = 'snc/data_protection/max' and not ( VALUE = '3' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/data_protection/min)" id="1.50.02c">
      <compliant>NAME = 'snc/data_protection/min' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'snc/data_protection/min' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/data_protection/use)" id="1.50.02d">
      <compliant>NAME = 'snc/data_protection/use' and VALUE = '3' </compliant>
      <noncompliant>NAME = 'snc/data_protection/use' and not( VALUE = '3' ) </noncompliant>
    </checkitem>
    <checkitem desc="Logon Ticket Issuing (login/accept_sso2_ticket)" id="1.50.03a">
      <compliant>NAME = 'login/accept_sso2_ticket' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'login/accept_sso2_ticket' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Logon Ticket Issuing (login/create_sso2_ticket)" id="1.50.03b">
      <compliant>NAME = 'login/create_sso2_ticket' and ( VALUE = '0' or VALUE = '3' )</compliant>
      <noncompliant>NAME = 'login/create_sso2_ticket' and not ( ( VALUE = '0' or VALUE = '3' ) ) </noncompliant>
    </checkitem>
    <checkitem desc="Logon Ticket Issuing (login/ticket_only_to_host)" id="1.50.03c">
      <compliant>NAME = 'login/ticket_only_to_host' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'login/ticket_only_to_host' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC authority check" id="1.60.02a">
      <compliant>NAME = 'auth/rfc_authority_check' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'auth/rfc_authority_check' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/acl_mode)" id="1.80.02a">
      <compliant>NAME = 'gw/acl_mode' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'gw/acl_mode' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/reg_no_conn_info)" id="1.80.02b">
      <compliant>NAME = 'gw/reg_no_conn_info' and VALUE = '129' </compliant>
      <noncompliant>NAME = 'gw/reg_no_conn_info' and not ( VALUE = '129' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/sec_info)" id="1.80.02c">
      <compliant>NAME = 'gw/sec_info' and VALUE like_regexpr '.*[Gg]lobal.*.secinfo.*' </compliant>
      <noncompliant>NAME = 'gw/sec_info' and not ( VALUE like_regexpr '.*[Gg]lobal.*.secinfo.*' ) </noncompliant>
    </checkitem>
    <!-- 1.80.03a rdisp/mssever <> _internal should be done with Bladelogic -->
    <checkitem desc="Message server hardening (ms/monitor)" id="1.80.03b">
      <compliant>NAME = 'ms/monitor' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'ms/monitor' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Message server hardening (ms/acl_info)" id="1.80.03c">
      <compliant>NAME = 'ms/acl_info' and VALUE like_regexpr '.*ms_acl_info' </compliant>
      <noncompliant>NAME = 'ms/acl_info' and not ( VALUE like_regexpr '.*ms_acl_info' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure access to SAP Management Console (service/protectedwebmethods)" id="1.80.07a">
      <compliant>NAME = 'service/protectedwebmethods' and VALUE = 'SDEFAULT' </compliant>
      <noncompliant>NAME = 'service/protectedwebmethods' and not ( VALUE = 'SDEFAULT' ) </noncompliant>
    </checkitem>
    <checkitem desc="Enforcement of HTTPS (icm/HTTP/redirect)" id="2.30.01a">
      <compliant>NAME = 'icm/HTTP/redirect_0' and VALUE like_regexpr '.*[Hh][Tt][Tt][Pp][Ss].*' </compliant>
      <noncompliant>NAME = 'icm/HTTP/redirect_0' and not ( VALUE like_regexpr '.*[Hh][Tt][Tt][Pp][Ss].*' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure session handling (icf/user_recheck)" id="2.60.01a">
      <compliant>NAME = 'icf/user_recheck' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'icf/user_recheck' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure cookie handling (login/ticket_only_by_https)" id="2.60.02a">
      <compliant>NAME = 'login/ticket_only_by_https' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'login/ticket_only_by_https' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure cookie handling (icf/set_HTTPonly_flag_on_cookies)" id="2.60.02b">
      <compliant>NAME = 'icf/set_HTTPonly_flag_on_cookies' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'icf/set_HTTPonly_flag_on_cookies' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Default logging mechanism (gw/logging)" id="2.70.01a">
      <compliant>NAME = 'gw/logging' and VALUE like_regexpr 'ACTION=SsPXM LOGFILE=.*gw_log.*' </compliant>
      <noncompliant>NAME = 'gw/logging' and not ( VALUE like_regexpr 'ACTION=SsPXM LOGFILE=.*gw_log.*' ) </noncompliant>
    </checkitem>
    <checkitem desc="Default logging mechanism (icm/HTTP/logging)" id="2.70.01b">
      <compliant>NAME = 'icm/HTTP/logging_0' and VALUE like_regexpr 'LOGFORMAT=SAPSMD2' </compliant>
      <noncompliant>NAME = 'icm/HTTP/logging_0' and not ( VALUE like_regexpr 'LOGFORMAT=SAPSMD2' ) </noncompliant>
    </checkitem>
    <checkitem desc="Activate table logging (rec/client)" id="2.70.02a">
      <compliant>NAME = 'rec/client' and ( VALUE like_regexpr '000' or VALUE like_regexpr 'ALL' )</compliant>
      <noncompliant>NAME = 'rec/client' and not ( ( VALUE like_regexpr '000' or VALUE like_regexpr 'ALL' ) ) </noncompliant>
    </checkitem>
    <checkitem desc="Create and document an authorization concept (rsau/enable) " id="2.80.01a">
      <compliant>NAME = 'rsau/enable' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'rsau/enable' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (login/min_password_diff)" id="2.90.01a">
      <compliant>NAME = 'login/min_password_diff' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_diff' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (login/min_password_specials)" id="2.90.01b">
      <compliant>NAME = 'login/min_password_specials' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_specials' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (login/password_max_idle_initial)" id="2.90.01c">
      <compliant>NAME = 'login/password_max_idle_initial' and rtrim(VALUE, ' -&gt;willbestbyABAPTC') between 1 and 7</compliant>
      <noncompliant>NAME = 'login/password_max_idle_initial' and not ( rtrim(VALUE, ' -&gt;willbestbyABAPTC') between 1 and 7 ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (login/password_max_idle_productive)" id="2.90.01d">
      <compliant>NAME = 'login/password_max_idle_productive' and VALUE between 1 and 90</compliant>
      <noncompliant>NAME = 'login/password_max_idle_productive' and not ( VALUE between 1 and 90 ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (login/fails_to_session_end)" id="2.90.01e">
      <compliant>NAME = 'login/fails_to_session_end' and VALUE = '3' </compliant>
      <noncompliant>NAME = 'login/fails_to_session_end' and not ( VALUE = '3' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (login/fails_to_user_lock)" id="2.90.01f">
      <compliant>NAME = 'login/fails_to_user_lock' and VALUE = '3' </compliant>
      <noncompliant>NAME = 'login/fails_to_user_lock' and not ( VALUE = '3' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (login/failed_user_auto_unlock)" id="2.90.01g">
      <compliant>NAME = 'login/failed_user_auto_unlock' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'login/failed_user_auto_unlock' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (is/HTTP/show_detailed_errors)" id="2.90.01h">
      <compliant>NAME = 'is/HTTP/show_detailed_errors' and VALUE like_regexpr '[Ff][Aa][Ll][Ss][Ee]' </compliant>
      <noncompliant>NAME = 'is/HTTP/show_detailed_errors' and not ( VALUE like_regexpr '[Ff][Aa][Ll][Ss][Ee]' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (is/HTTP/show_server_header)" id="2.90.01i">
      <compliant>NAME = 'is/HTTP/show_server_header' and VALUE like_regexpr '[Ff][Aa][Ll][Ss][Ee]' </compliant>
      <noncompliant>NAME = 'is/HTTP/show_server_header' and not ( VALUE like_regexpr '[Ff][Aa][Ll][Ss][Ee]' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (auth/object_disabling_active)" id="2.90.01j">
      <compliant>NAME = 'auth/object_disabling_active' and VALUE = 'N' </compliant>
      <noncompliant>NAME = 'auth/object_disabling_active' and not ( VALUE = 'N' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (ssl/ciphersuites)" id="2.90.01k">
      <compliant>NAME = 'ssl/ciphersuites' and VALUE = '135:HIGH:!eRC4:!eNULL:!aNULL:!mMD5' </compliant>
      <noncompliant>NAME = 'ssl/ciphersuites' and not ( VALUE = '135:HIGH:!eRC4:!eNULL:!aNULL:!mMD5' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (ssl/client_ciphersuites)" id="2.90.01l">
      <compliant>NAME = 'ssl/client_ciphersuites' and VALUE = '144:HIGH:!eRC4:!eNULL:!aNULL:!mMD5' </compliant>
      <noncompliant>NAME = 'ssl/client_ciphersuites' and not ( VALUE = '144:HIGH:!eRC4:!eNULL:!aNULL:!mMD5' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (system/secure_communication)" id="2.90.01m">
      <compliant>NAME = 'system/secure_communication' and VALUE = 'ON' </compliant>
      <noncompliant>NAME = 'system/secure_communication' and not ( VALUE = 'ON' ) </noncompliant>
    </checkitem>
    <checkitem desc="Set profile parameters (ixml/dtd_restriction)" id="2.90.01n">
      <compliant>NAME = 'ixml/dtd_restriction' and VALUE = 'prohibited' </compliant>
      <noncompliant>NAME = 'ixml/dtd_restriction' and not ( VALUE = 'prohibited' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/reg_info)" id="2.90.03a">
      <compliant>NAME = 'gw/reg_info' and VALUE like_regexpr '.*[Gg]lobal.*.reginfo.*' </compliant>
      <noncompliant>NAME = 'gw/reg_info' and not ( VALUE like_regexpr '.*[Gg]lobal.*.reginfo.*' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/prxy_info)" id="2.90.04a">
      <compliant>NAME = 'gw/prxy_info' and VALUE like_regexpr '.*[Gg]lobal.*.prxyinfo.*' </compliant>
      <noncompliant>NAME = 'gw/prxy_info' and not ( VALUE like_regexpr '.*[Gg]lobal.*.prxyinfo.*' ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for critical sicf services -->
  <configstore name="SICF_SERVICES" name_extended="" sci_id="">
    <checkitem desc="Deactivate critical ICF services (/sap/bc/soap/rfc)" id="1.80.05a" operator="not_exist:URL">
      <compliant>ICF_NAME = 'RFC' and URL = '/sap/bc/soap/rfc/' </compliant>
      <noncompliant>ICF_NAME = 'RFC' and URL = '/sap/bc/soap/rfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/echo)" id="1.80.05b" operator="not_exist:URL">
      <compliant>ICF_NAME = 'ECHO' and URL = '/sap/bc/echo/' </compliant>
      <noncompliant>ICF_NAME = 'ECHO' and URL = '/sap/bc/echo/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/FormToRfc)" id="1.80.05c" operator="not_exist:URL">
      <compliant>ICF_NAME = 'FORMTORFC' and URL = '/sap/bc/FormToRfc/' </compliant>
      <noncompliant>ICF_NAME = 'FORMTORFC' and URL = '/sap/bc/FormToRfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/report)" id="1.80.05d" operator="not_exist:URL">
      <compliant>ICF_NAME = 'REPORT' and URL = '/sap/bc/report/' </compliant>
      <noncompliant>ICF_NAME = 'REPORT' and URL = '/sap/bc/report/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/xrfc)" id="1.80.05e" operator="not_exist:URL">
      <compliant>ICF_NAME = 'XRFC' and URL = '/sap/bc/xrfc/' </compliant>
      <noncompliant>ICF_NAME = 'XRFC' and URL = '/sap/bc/xrfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/xrfc_test)" id="1.80.05f" operator="not_exist:URL">
      <compliant>ICF_NAME = 'XRFC_TEST' and URL = '/sap/bc/xrfc_test/' </compliant>
      <noncompliant>ICF_NAME = 'XRFC_TEST' and URL = '/sap/bc/xrfc_test/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/error)" id="1.80.05g" operator="not_exist:URL">
      <compliant>ICF_NAME = 'ERROR' and URL = '/sap/bc/error/' </compliant>
      <noncompliant>ICF_NAME = 'ERROR' and URL = '/sap/bc/error/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/webrfc)" id="1.80.05h" operator="not_exist:URL">
      <compliant>ICF_NAME = 'WEBRFC' and URL = '/sap/bc/webrfc/' </compliant>
      <noncompliant>ICF_NAME = 'WEBRFC' and URL = '/sap/bc/webrfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/certreq)" id="1.80.05i" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/bsp/sap/certreq/' </compliant>
      <noncompliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/bsp/sap/certreq/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/certmap)" id="1.80.05j" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/bsp/sap/certmap/' </compliant>
      <noncompliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/bsp/sap/certmap/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/gui/sap/its/CERTREQ)" id="1.80.05k" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/gui/sap/its/CERTREQ/' </compliant>
      <noncompliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/gui/sap/its/CERTREQ/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/gui/sap/its/CERTMAP)" id="1.80.05l" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/gui/sap/its/CERTMAP/' </compliant>
      <noncompliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/gui/sap/its/CERTMAP/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/bsp_veri)" id="1.80.05m" operator="not_exist:URL">
      <compliant>ICF_NAME = 'BSP_VERI' and URL = '/sap/bc/bsp/sap/bsp_veri/' </compliant>
      <noncompliant>ICF_NAME = 'BSP_VERI' and URL = '/sap/bc/bsp/sap/bsp_veri/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/icf)" id="1.80.05n" operator="not_exist:URL">
      <compliant>ICF_NAME = 'ICF' and URL = '/sap/bc/bsp/sap/icf' </compliant>
      <noncompliant>ICF_NAME = 'ICF' and URL = '/sap/bc/bsp/sap/icf' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/IDoc_XML)" id="1.80.05o" operator="not_exist:URL">
      <compliant>ICF_NAME = 'IDOC_XML' and URL = '/sap/bc/IDoc_XML' </compliant>
      <noncompliant>ICF_NAME = 'IDOC_XML' and URL = '/sap/bc/IDoc_XML' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/srt/IDoc)" id="1.80.05p" operator="not_exist:URL">
      <compliant>ICF_NAME = 'IDOC' and URL = '/sap/bc/srt/IDoc/' </compliant>
      <noncompliant>ICF_NAME = 'IDOC' and URL = '/sap/bc/srt/IDoc/' </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for security audit log -->
  <configstore name="AUDIT_CONFIGURATION">
    <checkitem desc="Default logging mechanism (Security Audit Log enabled)" id="2.70.01c">
      <compliant>name = 'Enabled' and value = 'Yes'</compliant>
      <noncompliant>name = 'Enabled'</noncompliant>
      <joinstore name="AUDIT_CONFIGURATION_SLOT" name_extended="" sci_id="">
        <joincompliant>SLOTINDEX like '%' and client =  '*'  and  msgvect like_regexpr '..........................................2.2.........................4.+' and selvar = '01' and status = 'X'</joincompliant>
        <joinnoncompliant>SLOTINDEX like '%'</joinnoncompliant>
      </joinstore>
    </checkitem>
  </configstore>
  <!-- Checks for standard users -->
  <configstore name="STANDARD_USERS" name_extended="" sci_id="">
    <checkitem desc="Secure SAP* (SAP* in client 000)" id="1.10.02b" operator="">
      <compliant>CLIENT = '000' and USERNAME = 'SAP*' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' and LOCKED = 'X' </compliant>
      <noncompliant>CLIENT = '000' and USERNAME = 'SAP*' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' and LOCKED = 'X' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure SAP* (SAP* in client 001)" id="1.10.02c" operator="">
      <compliant>CLIENT = '001' and USERNAME = 'SAP*' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' </compliant>
      <noncompliant>CLIENT = '001' and USERNAME = 'SAP*' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure SAP* (SAP* in client 066)" id="1.10.02d" operator="">
      <compliant>CLIENT = '066' and USERNAME = 'SAP*' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' </compliant>
      <noncompliant>CLIENT = '066' and USERNAME = 'SAP*' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure TMSADM (TMSADM in client 000)" id="1.10.03a" operator="">
      <compliant>CLIENT = '000' and USERNAME = 'TMSADM' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' ))</compliant>
      <noncompliant>CLIENT = '000' and USERNAME = 'TMSADM' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure EARLYWATCH, DDIC, and SAPCPIC (EARLYWATCH in 066)" id="1.10.04a" operator="">
      <compliant>CLIENT = '066' and USERNAME = 'EARLYWATCH' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and LOCKED = 'X' </compliant>
      <noncompliant>CLIENT = '066' and USERNAME = 'EARLYWATCH' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and LOCKED = 'X' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure EARLYWATCH, DDIC, and SAPCPIC (DDIC in 000)" id="1.10.04b" operator="">
      <compliant>CLIENT = '000' and USERNAME = 'DDIC' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' </compliant>
      <noncompliant>CLIENT = '000' and USERNAME = 'DDIC' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure EARLYWATCH, DDIC, and SAPCPIC (SAPCPIC in 000)" id="1.10.04c" operator="">
      <compliant>CLIENT = '000' and USERNAME = 'SAPCPIC' and ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' ) and LOCKED = 'X'  and EXISTING = 'X') or ( ( EXISTING = '' ) ) ) </compliant>
      <noncompliant>CLIENT = '000' and USERNAME = 'SAPCPIC' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN'  ) and LOCKED = 'X'  AND EXISTING = 'X' ) ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for password hashes -->
  <configstore name="USER_PASSWD_HASH_USAGE" name_extended="" sci_id="">
    <checkitem desc="Ensure usage of strong password hashes only (DIALOG users with BCODE exist)" id="1.20.02b" not_found="positive" operator="">
      <compliant>USER_TYPE = 'DIALOG'  and BCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'DIALOG'  and BCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SYSTEM users with BCODE exist)" id="1.20.02c" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SYSTEM'  and BCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'SYSTEM'  and BCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (COMM users with BCODE exist)" id="1.20.02d" not_found="positive" operator="">
      <compliant>USER_TYPE = 'COMMUNICATION'  and BCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'COMMUNICATION'  and BCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SERVICE users with BCODE exist)" id="1.20.02e" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SERVICE'  and BCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'SERVICE'  and BCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (DIALOG users with PASSCODE exist)" id="1.20.02f" not_found="positive" operator="">
      <compliant>USER_TYPE = 'DIALOG'  and PASSCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'DIALOG'  and PASSCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SYSTEM users with PASSCODE exist)" id="1.20.02g" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SYSTEM'  and PASSCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'SYSTEM'  and PASSCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (COMM users with PASSCODE exist)" id="1.20.02h" not_found="positive" operator="">
      <compliant>USER_TYPE = 'COMMUNICATION'  and PASSCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'COMMUNICATION'  and PASSCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SERVICE users with PASSCODE exist)" id="1.20.02i" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SERVICE'  and PASSCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'SERVICE'  and PASSCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for system modifiable setting -->
  <configstore name="GLOBAL" name_extended="" sci_id="">
    <checkitem desc="System and Client Change options (System modifiable)" id="2.90.02a" operator="">
      <compliant>NAME = 'GLOBAL_SETTING'  and VALUE &lt;&gt; 'MODIFIABLE' </compliant>
      <noncompliant>NAME = 'GLOBAL_SETTING'  and not ( VALUE &lt;&gt; 'MODIFIABLE'  ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for gw secinfo -->
  <configstore name="GW_SECINFO" name_extended="" sci_id="">
    <checkitem desc="RFC gateway hardening (secinfo content)" id="1.80.02d" operator="">
      <compliant>line like '%' and ( (text like_regexpr 'USER-HOST=[^*]' and text like_regexpr '[^-]HOST=[^*]') or text like_regexpr 'HOST=[*][.]')</compliant>
      <noncompliant>line like '%' and not ( (text like_regexpr 'USER-HOST=[^*]' and text like_regexpr '[^-]HOST=[^*]') or text like_regexpr 'HOST=[*][.]')</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for ms acl info -->
  <configstore name="MS_SECINFO" name_extended="" sci_id="">
    <checkitem desc="Message server hardening (acl_info content)" id="1.80.03d" operator="">
      <compliant>line like '%' and text like_regexpr 'HOST=[^*]'</compliant>
      <noncompliant>line like '%' and text = 'HOST=*'</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for table authorization groups -->
  <configstore name="TDDAT_TABLES" name_extended="" sci_id="">
    <checkitem desc="Secure cryptographic keys" id="1.80.04a" not_found="positive" operator="">
      <compliant>TABNAME = 'SSF_PSE_D'  and CCLASS = 'SPSE' </compliant>
      <noncompliant>TABNAME = 'SSF_PSE_D'  and not ( CCLASS = 'SPSE'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USR02)" id="1.20.03a" not_found="positive" operator="">
      <compliant>TABNAME = 'USR02'  and CCLASS = 'SPWD' </compliant>
      <noncompliant>TABNAME = 'USR02'  and not ( CCLASS = 'SPWD'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USH02)" id="1.20.03b" not_found="positive" operator="">
      <compliant>TABNAME = 'USH02'  and CCLASS = 'SPWD' </compliant>
      <noncompliant>TABNAME = 'USH02'  and not ( CCLASS = 'SPWD'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (VUSER001)" id="1.20.03c" not_found="positive" operator="">
      <compliant>TABNAME = 'VUSER001'  and CCLASS = 'SPWD' </compliant>
      <noncompliant>TABNAME = 'VUSER001'  and not ( CCLASS = 'SPWD'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USH02_ARC_TMP)" id="1.20.03d" not_found="positive" operator="">
      <compliant>TABNAME = 'USH02_ARC_TMP'  and CCLASS = 'SPWD' </compliant>
      <noncompliant>TABNAME = 'USH02_ARC_TMP'  and not ( CCLASS = 'SPWD'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USRPWDHISTORY)" id="1.20.03e" not_found="positive" operator="">
      <compliant>TABNAME = 'USRPWDHISTORY'  and CCLASS = 'SPWD' </compliant>
      <noncompliant>TABNAME = 'USRPWDHISTORY'  and not ( CCLASS = 'SPWD'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (VUSR02_PWD)" id="1.20.03f" not_found="positive" operator="">
      <compliant>TABNAME = 'VUSR02_PWD'  and CCLASS = 'SPWD' </compliant>
      <noncompliant>TABNAME = 'VUSR02_PWD'  and not ( CCLASS = 'SPWD'  ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for session management -->
  <configstore name="SESSION_MANAGEMENT" name_extended="" sci_id="">
    <checkitem desc="Secure session handling" id="2.60.01b" operator="">
      <compliant>NAME = 'SESSION_MANAGEMENT'  and VALUE = 'ACTIVE' </compliant>
      <noncompliant>NAME = 'SESSION_MANAGEMENT'  and not ( VALUE = 'ACTIVE'  ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for authorization profiles -->
  <configstore name="AUTH_PROFILE_USER" name_extended="" sci_id="">
    <checkitem desc="Create and document an authorization concept (user with profile SAP_ALL exists)" id="2.80.01b" operator="">
      <compliant>PROFILE = 'SAP_ALL' and USERNAME in ('BATCH_USER','DDIC','SAP*','CAM_FALLBACK','CAM_CHANGE')  and STATUS = 'USER_IS_AUTHORIZED' </compliant>
      <noncompliant>PROFILE = 'SAP_ALL' and USERNAME not in ('BATCH_USER','DDIC','SAP*','CAM_FALLBACK','CAM_CHANGE') and  STATUS = 'USER_IS_AUTHORIZED'</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for gw reginfo -->
  <configstore name="GW_REGINFO" name_extended="" sci_id="">
    <checkitem desc="RFC gateway hardening (reginfo content)" id="2.90.03b" operator="">
      <compliant>line like '%' and ( TEXT like_regexpr 'HOST=[^*]' )</compliant>
      <noncompliant> (line like '%' and TEXT like  '%HOST=[*]%' ) OR ( line like '%' and TEXT = 'P TP=*' )</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for gw prxyinfo -->
  <configstore name="GW_PRXINFO" name_extended="" sci_id="">
    <checkitem desc="RFC gateway hardening (prxyinfo content)" id="2.90.04b" operator="">
      <compliant>line like '%' and ( text like  '%P SOURCE=local,internal%' or text like '%SOURCE=*%'  )</compliant>
      <noncompliant>line like '%' and ( not ( text like  '%P SOURCE=local,internal%' or text like '%SOURCE=*%' ) or text = 'gw/prxy_info file not found')</noncompliant>
    </checkitem>
  </configstore>
</targetsystem>