<?xml version="1.0" encoding="utf-8"?>
<!-- Intention:
Check OS version
 -->
<!-- 0001 / MODIFIED NUMBER VALUES TO RANGE -->
<targetsystem desc="BSO_OS_VER" id="BSO_OS_VER" multisql="Yes">
  <configstore name="SAP_ITSAMOperatingSystem">
    <checkitem desc="operating system check" id="1.0.0.0">
      <compliant>name = 'PPMSRelease' and VALUE = 'SLES11' or VALUE = 'SLES12'</compliant>
      <noncompliant>name = 'PPMSRelease' and not (VALUE =  'SLES11' or VALUE = 'SLES12')</noncompliant>
    </checkitem>
  </configstore>
</targetsystem>