<?xml version="1.0" encoding="utf-16"?>
<!-- Author: D038680
Version: 20160609
20170614 d02470 checks 2.90.01g,2.90.01h,2.90.01i,2.90.01j,2.90.02a,2.90.02b.2.90.02c using config store parameters
20170511 d024870 removed "-optional" from check ids, causing errors in extractors of Jens Leuschner
20160609 d038680, added system_type="JAVA" to configstores servlet_jsp and ume.service for compatibility with SRSM
20160510 d038680, new checks for default accounts 1.10.01a, 1.10.01b
20160509 d038680, removed configstore http.properties because of incompatibility. 
	Store not available in SRSM with Java 7.40 systems. Causing errors in SRSM 
20160427 d038680, new precheck 0.00.01
20160128 d038680, initial version
-->
<targetsystem desc="SEC_JAVA_BASELINE" id="SEC_JAVA_BASELINE" multisql="Yes">
  <!-- Checks for Software Conponents -->
  <configstore name="COMP_LEVEL" name_extended="" system_type="JAVA">
    <checkitem desc="Precheck: LM-TOOLS exists" id="0.00.01" operator="exist:COMPONENT">
      <compliant>COMPONENT = 'LM-TOOLS' </compliant>
      <noncompliant>COMPONENT = 'LM-TOOLS' </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for default users -->
  <configstore name="sapUserProperties:Guest" name_extended="" system_type="JAVA">
    <checkitem desc="Secure default users (Guest locked)" id="1.10.01a" not_found="positive" operator="">
      <compliant>ID = 'USER.PRIVATE_DATASOURCE.un:Guest' and IS_LOCKED = 'true'</compliant>
      <noncompliant>ID = 'USER.PRIVATE_DATASOURCE.un:Guest' and not ( IS_LOCKED = 'true' )</noncompliant>
    </checkitem>
  </configstore>
  <configstore name="sapUserProperties:J2EE_GUEST" name_extended="" system_type="JAVA">
    <checkitem desc="Secure default users (J2EE_GUEST locked)" id="1.10.01b" not_found="positive" operator="">
      <compliant>ID = 'USER.PRIVATE_DATASOURCE.un:J2EE_GUEST' and IS_LOCKED = 'true'</compliant>
      <noncompliant>ID = 'USER.PRIVATE_DATASOURCE.un:J2EE_GUEST' and not ( IS_LOCKED = 'true' )</noncompliant>
    </checkitem>
  </configstore>
  <configstore name="sapUserProperties:Administrator" name_extended="" system_type="JAVA">
    <checkitem desc="Secure default users (Administrator locked)" id="1.10.01c" not_found="positive" operator="">
      <compliant>ID = 'USER.PRIVATE_DATASOURCE.un:Administrator' and IS_LOCKED = 'true'</compliant>
      <noncompliant>ID = 'USER.PRIVATE_DATASOURCE.un:Administrator' and not ( IS_LOCKED = 'true' )</noncompliant>
    </checkitem>
  </configstore>
  <configstore name="servlet_jsp" name_extended="" sci_id="" system_type="JAVA">
    <checkitem desc="Disable invoker servlet" id="1.80.05a" operator="">
      <compliant>NAME = 'EnableInvokerServletGlobally'  and VALUE = 'false' </compliant>
      <noncompliant>NAME = 'EnableInvokerServletGlobally'  and not ( VALUE = 'false'  ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for UME properties -->
  <configstore name="com.sap.security.core.ume.service" name_extended="" sci_id="" system_type="JAVA">
    <checkitem desc="Configure a compliant password policy (min_length)" id="1.20.01a" operator="">
      <compliant>NAME = 'ume.logon.security_policy.password_min_length'  and VALUE &gt;= '8' </compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.password_min_length'  and not ( VALUE &gt;= '8'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (alpha_numeric_required)" id="1.20.01b" operator="">
      <compliant>NAME = 'ume.logon.security_policy.password_alpha_numeric_required'  and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.password_alpha_numeric_required'  and not ( VALUE &gt;= '1'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (mix_case_required)" id="1.20.01c" operator="">
      <compliant>NAME = 'ume.logon.security_policy.password_mix_case_required'  and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.password_mix_case_required'  and not ( VALUE &gt;= '1'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (expire_days)" id="1.20.01d" operator="">
      <compliant>NAME = 'ume.logon.security_policy.password_expire_days'  and replace_regexpr ('^$' IN VALUE WITH 0) between 1 and 182</compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.password_expire_days'  and not ( replace_regexpr ('^$' IN VALUE WITH 0) between 1 and 182 ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (history)" id="1.20.01e" operator="">
      <compliant>NAME = 'ume.logon.security_policy.password_history'  and VALUE &gt;= '5' </compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.password_history'  and not ( VALUE &gt;= '5'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (enforce_policy_at_logon)" id="1.20.01f" operator="">
      <compliant>NAME = 'ume.logon.security_policy.enforce_policy_at_logon'  and VALUE = 'true' </compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.enforce_policy_at_logon'  and not ( VALUE = 'true'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (special character required)" id="1.20.01g" operator="">
      <compliant>NAME = 'ume.logon.security_policy.password_special_char_required'  and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.password_special_char_required' and not ( VALUE &gt;= '1'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (Max Password idle time)" id="1.20.01h" operator="">
      <compliant>NAME = 'ume.logon.security_policy.password_max_idle_time'  and VALUE &gt;= '90' </compliant>
      <noncompliant>NAME = 'ume.logon.security_policy.password_max_idle_time' and not ( VALUE &gt;= '90'  ) </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>