<?xml version="1.0" encoding="utf-16"?>
<!-- Changes:
 -->
<targetsystem desc="SEC_ABAP_BASELINE" id="SEC_ABAP_BASELINE" multisql="Yes">
  <!-- 
Checks for client settings 
I7D/002 is a productive client used by PPMS development -> Exception
ESD/100 is a productive client used by AMS development -> Exception
-->
  <configstore name="CLIENTS" name_extended="" sci_id="">
    <checkitem desc="Delete unused clients (except 001 and 000 and known exception)" id="1.10.01b" not_found="positive">
      <compliant>MANDT like '%' and not MANDT = '001' and not MANDT = '000'</compliant>
      <noncompliant>MANDT like '%' and not MANDT = '001' and not MANDT = '000'</noncompliant>
      <joinstore name="ENV_VARIABLES">
        <joincompliant> NAME = 'SAPSYSTEMNAME' and (VALUE = 'I7D' or VALUE='ESD')</joincompliant>
        <joinnoncompliant> NAME = 'SAPSYSTEMNAME' and NOT (VALUE = 'I7D' or VALUE='ESD')</joinnoncompliant>
      </joinstore>
    </checkitem>
  </configstore>
  <!-- Checks for profile parametes -->
  <configstore name="ABAP_INSTANCE_PAHI" name_extended="" sci_id="">
    <checkitem desc="Secure SAP* (login/no_automatic_user_sapstar)" id="1.10.02a">
      <compliant>NAME = 'login/no_automatic_user_sapstar' and VALUE = '1' </compliant>
      <complianttext>The SAP* account is secured with parameter login/no_automatic_user_sapstar = 1</complianttext>
      <noncompliant>NAME = 'login/no_automatic_user_sapstar' and not ( VALUE = '1' ) </noncompliant>
      <noncomplianttext>The SAP* account MUST be secured with parameter login/no_automatic_user_sapstar = 1'</noncomplianttext>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_lng)" id="1.20.01a">
      <compliant>NAME = 'login/min_password_lng' and VALUE &gt;= '8' </compliant>
      <complianttext>A compliant password policy (login/min_password_lng &gt;= 8 is configured</complianttext>
      <noncompliant>NAME = 'login/min_password_lng' and not ( VALUE &gt;= '8' ) </noncompliant>
      <noncomplianttext>Configure a compliant password policy (login/min_password_lng &gt;= 8)</noncomplianttext>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_letters)" id="1.20.01b">
      <compliant>NAME = 'login/min_password_letters' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_letters' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_digits)" id="1.20.01c">
      <compliant>NAME = 'login/min_password_digits' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_digits' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_lowercase)" id="1.20.01d">
      <compliant>NAME = 'login/min_password_lowercase' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_lowercase' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/min_password_uppercase)" id="1.20.01e">
      <compliant>NAME = 'login/min_password_uppercase' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/min_password_uppercase' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_expiration_time)" id="1.20.01f">
      <compliant>NAME = 'login/password_expiration_time' and replace(VALUE, ' (not 0)' , '9999') between '1' and '182'</compliant>
      <noncompliant>NAME = 'login/password_expiration_time' and not ( replace(VALUE, ' (not 0)' , '9999') between '1' and '182' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_history_size)" id="1.20.01g">
      <compliant>NAME = 'login/password_history_size' and VALUE &gt;= '5' </compliant>
      <noncompliant>NAME = 'login/password_history_size' and not ( VALUE &gt;= '5' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_change_waittime)" id="1.20.01h">
      <compliant>NAME = 'login/password_change_waittime' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'login/password_change_waittime' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure a compliant password policy (login/password_compliance_to_current_policy)" id="1.20.01i">
      <compliant>NAME = 'login/password_compliance_to_current_policy' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'login/password_compliance_to_current_policy' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (login/password_downwards_compatibility)" id="1.20.02a">
      <compliant>NAME = 'login/password_downwards_compatibility' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'login/password_downwards_compatibility' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/enable)" id="1.50.02a">
      <compliant>NAME = 'snc/enable' and VALUE like '1%' </compliant>
      <noncompliant>NAME = 'snc/enable' and not VALUE like '1%' </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/data_protection/max)" id="1.50.02b">
      <compliant>NAME = 'snc/data_protection/max' and VALUE = '3' </compliant>
      <noncompliant>NAME = 'snc/data_protection/max' and not ( VALUE = '3' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/data_protection/min)" id="1.50.02c">
      <compliant>NAME = 'snc/data_protection/min' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'snc/data_protection/min' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/data_protection/use)" id="1.50.02d">
      <compliant>NAME = 'snc/data_protection/use' and VALUE = '3' </compliant>
      <noncompliant>NAME = 'snc/data_protection/use' and not( VALUE = '3' ) </noncompliant>
    </checkitem>
    <checkitem desc="Configure SNC from SAP GUI to SAP system (snc/accept_insecure_gui)" id="1.50.02e">
      <compliant>NAME = 'snc/accept_insecure_gui' and VALUE = 'U' </compliant>
      <noncompliant>NAME = 'snc/accept_insecure_gui' and not( VALUE = 'U' ) </noncompliant>
    </checkitem>
    <checkitem desc="Logon Ticket Issuing (login/accept_sso2_ticket)" id="1.50.03a">
      <compliant>NAME = 'login/accept_sso2_ticket' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'login/accept_sso2_ticket' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Logon Ticket Issuing (login/create_sso2_ticket)" id="1.50.03b">
      <compliant>NAME = 'login/create_sso2_ticket' and ( VALUE = '0' or VALUE = '3' )</compliant>
      <noncompliant>NAME = 'login/create_sso2_ticket' and not ( ( VALUE = '0' or VALUE = '3' ) ) </noncompliant>
    </checkitem>
    <checkitem desc="Logon Ticket Issuing (login/ticket_only_to_host)" id="1.50.03c">
      <compliant>NAME = 'login/ticket_only_to_host' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'login/ticket_only_to_host' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC authority check" id="1.60.02a">
      <compliant>NAME = 'auth/rfc_authority_check' and VALUE &gt;= '1' </compliant>
      <noncompliant>NAME = 'auth/rfc_authority_check' and not ( VALUE &gt;= '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/acl_mode)" id="1.80.02a">
      <compliant>NAME = 'gw/acl_mode' and VALUE = '1' </compliant>
      <noncompliant>NAME = 'gw/acl_mode' and not ( VALUE = '1' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/reg_no_conn_info)" id="1.80.02b">
      <compliant>NAME = 'gw/reg_no_conn_info' and VALUE = '129' </compliant>
      <noncompliant>NAME = 'gw/reg_no_conn_info' and not ( VALUE = '129' ) </noncompliant>
    </checkitem>
    <checkitem desc="RFC gateway hardening (gw/sec_info)" id="1.80.02c">
      <compliant>NAME = 'gw/sec_info' and (VALUE like_regexpr '.*secinfo.*' or VALUE like_regexpr '.*SYS.*.secinfo.*') </compliant>
      <noncompliant>NAME = 'gw/sec_info' and not ( VALUE like_regexpr '.*secinfo.*' or VALUE like_regexpr '.*SYS.*.secinfo.*')</noncompliant>
    </checkitem>
    <!-- 1.80.03a rdisp/mssever <> _internal should be done with Bladelogic -->
    <checkitem desc="Message server hardening (ms/monitor)" id="1.80.03b">
      <compliant>NAME = 'ms/monitor' and VALUE = '0' </compliant>
      <noncompliant>NAME = 'ms/monitor' and not ( VALUE = '0' ) </noncompliant>
    </checkitem>
    <checkitem desc="Message server hardening (ms/acl_info)" id="1.80.03c">
      <compliant>NAME = 'ms/acl_info' and VALUE like_regexpr '.*ms_acl_info' </compliant>
      <noncompliant>NAME = 'ms/acl_info' and not ( VALUE like_regexpr '.*ms_acl_info' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure access to SAP Management Console (service/protectedwebmethods)" id="1.80.07a">
      <compliant>NAME = 'service/protectedwebmethods' and (VALUE = 'SDEFAULT')</compliant>
      <noncompliant>NAME = 'service/protectedwebmethods' and not (VALUE = 'SDEFAULT')</noncompliant>
    </checkitem>
    <checkitem desc="Secure access to transaction codes" id="1.80.08a">
      <compliant>NAME = 'auth/tcodes_not_checked' and (VALUE = 'N' or VALUE = '')</compliant>
      <noncompliant>NAME = 'auth/tcodes_not_checked' and not (VALUE = 'N' or VALUE = '')</noncompliant>
    </checkitem>
    <checkitem desc="Secure multiple SAPGUI logons" id="1.80.09a">
      <compliant>NAME = 'login/disable_multi_gui_login' and (VALUE = '1')</compliant>
      <noncompliant>NAME = 'login/disable_multi_gui_login' and not (VALUE = '1')</noncompliant>
    </checkitem>
    <checkitem desc="Maximum timeout of no input at GUI" id="1.80.10a">
      <compliant>NAME = 'rdisp/gui_auto_logout' and (VALUE &lt;= '7200')</compliant>
      <noncompliant>NAME = 'rdisp/gui_auto_logout' and (VALUE &gt; '7200')</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for critical sicf services -->
  <configstore name="SICF_SERVICES" name_extended="" sci_id="">
    <checkitem desc="Deactivate critical ICF services (/sap/bc/soap/rfc)" id="1.80.05a" operator="not_exist:URL">
      <compliant>ICF_NAME = 'RFC' and URL = '/sap/bc/soap/rfc/' </compliant>
      <noncompliant>ICF_NAME = 'RFC' and URL = '/sap/bc/soap/rfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/echo)" id="1.80.05b" operator="not_exist:URL">
      <compliant>ICF_NAME = 'ECHO' and URL = '/sap/bc/echo/' </compliant>
      <noncompliant>ICF_NAME = 'ECHO' and URL = '/sap/bc/echo/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/FormToRfc)" id="1.80.05c" operator="not_exist:URL">
      <compliant>ICF_NAME = 'FORMTORFC' and URL = '/sap/bc/FormToRfc/' </compliant>
      <noncompliant>ICF_NAME = 'FORMTORFC' and URL = '/sap/bc/FormToRfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/report)" id="1.80.05d" operator="not_exist:URL">
      <compliant>ICF_NAME = 'REPORT' and URL = '/sap/bc/report/' </compliant>
      <noncompliant>ICF_NAME = 'REPORT' and URL = '/sap/bc/report/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/xrfc)" id="1.80.05e" operator="not_exist:URL">
      <compliant>ICF_NAME = 'XRFC' and URL = '/sap/bc/xrfc/' </compliant>
      <noncompliant>ICF_NAME = 'XRFC' and URL = '/sap/bc/xrfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/xrfc_test)" id="1.80.05f" operator="not_exist:URL">
      <compliant>ICF_NAME = 'XRFC_TEST' and URL = '/sap/bc/xrfc_test/' </compliant>
      <noncompliant>ICF_NAME = 'XRFC_TEST' and URL = '/sap/bc/xrfc_test/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/error)" id="1.80.05g" operator="not_exist:URL">
      <compliant>ICF_NAME = 'ERROR' and URL = '/sap/bc/error/' </compliant>
      <noncompliant>ICF_NAME = 'ERROR' and URL = '/sap/bc/error/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/webrfc)" id="1.80.05h" operator="not_exist:URL">
      <compliant>ICF_NAME = 'WEBRFC' and URL = '/sap/bc/webrfc/' </compliant>
      <noncompliant>ICF_NAME = 'WEBRFC' and URL = '/sap/bc/webrfc/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/certreq)" id="1.80.05i" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/bsp/sap/certreq/' </compliant>
      <noncompliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/bsp/sap/certreq/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/certmap)" id="1.80.05j" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/bsp/sap/certmap/' </compliant>
      <noncompliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/bsp/sap/certmap/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/gui/sap/its/CERTREQ)" id="1.80.05k" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/gui/sap/its/CERTREQ/' </compliant>
      <noncompliant>ICF_NAME = 'CERTREQ' and URL = '/sap/bc/gui/sap/its/CERTREQ/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/gui/sap/its/CERTMAP)" id="1.80.05l" operator="not_exist:URL">
      <compliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/gui/sap/its/CERTMAP/' </compliant>
      <noncompliant>ICF_NAME = 'CERTMAP' and URL = '/sap/bc/gui/sap/its/CERTMAP/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/bsp_veri)" id="1.80.05m" operator="not_exist:URL">
      <compliant>ICF_NAME = 'BSP_VERI' and URL = '/sap/bc/bsp/sap/bsp_veri/' </compliant>
      <noncompliant>ICF_NAME = 'BSP_VERI' and URL = '/sap/bc/bsp/sap/bsp_veri/' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/bsp/sap/icf)" id="1.80.05n" operator="not_exist:URL">
      <compliant>ICF_NAME = 'ICF' and URL = '/sap/bc/bsp/sap/icf' </compliant>
      <noncompliant>ICF_NAME = 'ICF' and URL = '/sap/bc/bsp/sap/icf' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/IDoc_XML)" id="1.80.05o" operator="not_exist:URL">
      <compliant>ICF_NAME = 'IDOC_XML' and URL = '/sap/bc/IDoc_XML' </compliant>
      <noncompliant>ICF_NAME = 'IDOC_XML' and URL = '/sap/bc/IDoc_XML' </noncompliant>
    </checkitem>
    <checkitem desc="Deactivate critical ICF services (/sap/bc/srt/IDoc)" id="1.80.05p" operator="not_exist:URL">
      <compliant>ICF_NAME = 'IDOC' and URL = '/sap/bc/srt/IDoc/' </compliant>
      <noncompliant>ICF_NAME = 'IDOC' and URL = '/sap/bc/srt/IDoc/' </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for standard users -->
  <configstore name="STANDARD_USERS" name_extended="" sci_id="">
    <checkitem desc="Secure SAP* (SAP* in client 000)" id="1.10.02b" operator="">
      <compliant>CLIENT = '000' and USERNAME = 'SAP*' and ((PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = 'X') OR (PASSWORD_STATUS = '' AND EXISTING = '' AND LOCKED = ''))</compliant>
      <noncompliant>CLIENT = '000' and USERNAME = 'SAP*' and ((PASSWORD_STATUS = 'DEFAULT' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'DEFAULT' and EXISTING = '' AND LOCKED = ''))</noncompliant>
    </checkitem>
    <checkitem desc="Secure SAP* (SAP* in client 001)" id="1.10.02c" operator="not_exist:CLIENT">
      <compliant>CLIENT = '001' and USERNAME = 'SAP*' and ((PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = 'X') OR (PASSWORD_STATUS = '' AND EXISTING = '' AND LOCKED = ''))</compliant>
      <noncompliant>CLIENT = '001' and USERNAME = 'SAP*' and ((PASSWORD_STATUS = 'DEFAULT' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'DEFAULT' and EXISTING = '' AND LOCKED = ''))</noncompliant>
    </checkitem>
    <checkitem desc="Secure SAP* (SAP* in client 066 if exist)" id="1.10.02e" not_found="positive" operator="">
      <compliant>CLIENT = '066' and USERNAME = 'SAP*' and ((PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = 'X') OR (PASSWORD_STATUS = '' AND EXISTING = '' AND LOCKED = ''))</compliant>
      <noncompliant>CLIENT = '066' and USERNAME = 'SAP*' and ((PASSWORD_STATUS = 'DEFAULT' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'DEFAULT' and EXISTING = '' AND LOCKED = ''))</noncompliant>
    </checkitem>
    <checkitem desc="Secure SAP* (SAP* in ALL clients)" id="1.10.02f" not_found="positive" operator="not_exist:CLIENT">
      <compliant>CLIENT like '%' and not CLIENT = '001' and not CLIENT = '000' and not CLIENT = '066' and USERNAME = 'SAP*' and (USERNAME = 'SAP*' and (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = 'X') OR (PASSWORD_STATUS = '' AND EXISTING = '' AND LOCKED = ''))</compliant>
      <noncompliant>CLIENT like '%' and not CLIENT = '001' and not CLIENT = '000' and not CLIENT = '066' and USERNAME = 'SAP*' and (USERNAME = 'SAP*' and (PASSWORD_STATUS = 'DEFAULT' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'DEFAULT' and EXISTING = '' AND LOCKED = ''))</noncompliant>
    </checkitem>
    <checkitem desc="Secure TMSADM (TMSADM in client 000)" id="1.10.03a" operator="">
      <compliant>CLIENT = '000' and USERNAME = 'TMSADM' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' ))</compliant>
      <noncompliant>CLIENT = '000' and USERNAME = 'TMSADM' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure EARLYWATCH, DDIC, and SAPCPIC (EARLYWATCH in all clients)" id="1.10.04a" not_found="positive" operator="">
      <compliant>CLIENT like '%' and USERNAME = 'EARLYWATCH' and ((PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = 'X') OR (PASSWORD_STATUS = '' AND EXISTING = '' AND LOCKED = ''))</compliant>
      <noncompliant>CLIENT like '%' and USERNAME = 'EARLYWATCH' and ((PASSWORD_STATUS = 'DEFAULT' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'DEFAULT' and EXISTING = '' AND LOCKED = ''))         </noncompliant>
    </checkitem>
    <checkitem desc="Secure EARLYWATCH, DDIC, and SAPCPIC (DDIC in 000)" id="1.10.04b" operator="">
      <compliant>CLIENT = '000' and USERNAME = 'DDIC' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' ) and EXISTING = 'X')</compliant>
      <noncompliant>CLIENT = '000' and USERNAME = 'DDIC' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure EARLYWATCH, DDIC, and SAPCPIC (SAPCPIC in all clients)" id="1.10.04c" not_found="positive" operator="">
      <compliant>CLIENT like '%' and USERNAME = 'SAPCPIC' and ((PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = 'X') OR (PASSWORD_STATUS = '' AND EXISTING = '' AND LOCKED = ''))</compliant>
      <noncompliant>CLIENT like '%' and USERNAME = 'SAPCPIC' and ((PASSWORD_STATUS = 'DEFAULT' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'DEFAULT' and EXISTING = '' AND LOCKED = ''))         </noncompliant>
    </checkitem>
    <checkitem desc="Secure DDIC (DDIC in client 001)" id="1.10.04d" operator="" system_roles="PROD">
      <compliant>USERNAME = 'DDIC' and not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' and LOCKED = 'X' </compliant>
      <noncompliant>CLIENT = '001' and USERNAME = 'DDIC' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' and LOCKED = 'X' ) </noncompliant>
    </checkitem>
    <checkitem desc="Secure DDIC in all clients except 000" id="1.10.04e" operator="not_exist:CLIENT">
      <compliant>CLIENT like '%' and not CLIENT = '000' and not CLIENT = '001' and USERNAME = 'DDIC' and (USERNAME = 'DDIC' and (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = 'X') OR (PASSWORD_STATUS = '' AND EXISTING = '' AND LOCKED = ''))</compliant>
      <noncompliant>CLIENT like '%' and not CLIENT = '000' and not CLIENT = '001' and USERNAME = 'DDIC' and (USERNAME = 'DDIC' and (PASSWORD_STATUS = 'DEFAULT' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'CHANGED' AND EXISTING = 'X' AND LOCKED = '') OR (PASSWORD_STATUS = 'DEFAULT' and EXISTING = '' AND LOCKED = ''))</noncompliant>
    </checkitem>
    <checkitem desc="Secure ALL standard users in all remaining clients" id="1.10.04f" not_found="positive" operator="">
      <compliant>USERNAME like '%' and CLIENT like '%' and CLIENT &lt;&gt; '066' and CLIENT &lt;&gt; '001' and CLIENT &lt;&gt; '000' and not USERNAME = 'SAP*' and not ( (PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' and LOCKED = 'X' </compliant>
      <noncompliant>USERNAME like '%' and CLIENT like '%' and CLIENT &lt;&gt; '066' and CLIENT &lt;&gt; '001' and CLIENT &lt;&gt; '000' and not USERNAME = 'SAP*' and PASSWORD_STATUS = 'DEFAULT' and not ( not ( ( PASSWORD_STATUS = 'DEFAULT' or PASSWORD_STATUS = 'UNKNOWN' )) and EXISTING = 'X' and LOCKED = 'X' ) </noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for password hashes -->
  <configstore name="USER_PASSWD_HASH_USAGE" name_extended="" sci_id="">
    <checkitem desc="Ensure usage of strong password hashes only (DIALOG users with BCODE exist)" id="1.20.02b" not_found="positive" operator="">
      <compliant>USER_TYPE = 'DIALOG'  and BCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'DIALOG'  and BCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SYSTEM users with BCODE exist)" id="1.20.02c" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SYSTEM'  and BCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'SYSTEM'  and BCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (COMM users with BCODE exist)" id="1.20.02d" not_found="positive" operator="">
      <compliant>USER_TYPE = 'COMMUNICATION'  and BCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'COMMUNICATION'  and BCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SERVICE users with BCODE exist)" id="1.20.02e" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SERVICE'  and BCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'SERVICE'  and BCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (DIALOG users with PASSCODE exist)" id="1.20.02f" not_found="positive" operator="">
      <compliant>USER_TYPE = 'DIALOG'  and PASSCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'DIALOG'  and PASSCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SYSTEM users with PASSCODE exist)" id="1.20.02g" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SYSTEM'  and PASSCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'SYSTEM'  and PASSCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (COMM users with PASSCODE exist)" id="1.20.02h" not_found="positive" operator="">
      <compliant>USER_TYPE = 'COMMUNICATION'  and PASSCODE = 'USED'  and NUM_USER &lt;= '0' </compliant>
      <noncompliant>USER_TYPE = 'COMMUNICATION'  and PASSCODE = 'USED'  and not ( NUM_USER &lt;= '0'  ) </noncompliant>
    </checkitem>
    <checkitem desc="Ensure usage of strong password hashes only (SERVICE users with PASSCODE exist)" id="1.20.02i" not_found="positive" operator="">
      <compliant>USER_TYPE = 'SERVICE'  and PASSCODE = 'USED'  and NUM_USER = '0' </compliant>
      <noncompliant>USER_TYPE = 'SERVICE'  and PASSCODE = 'USED'  and not ( NUM_USER = '0'  ) </noncompliant>
    </checkitem>
  </configstore>
  <configstore name="AUTH_PROFILE_USER" name_extended="" sci_id="">
    <checkitem desc="DDIC MUST be type DIALOG" id="1.110.01a" operator="">
      <compliant>USERNAME = 'DDIC' and USER_TYPE = 'DIALOG'</compliant>
      <noncompliant>USERNAME = 'DDIC' and NOT (USER_TYPE = 'DIALOG')</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for gw secinfo -->
  <configstore name="GW_SECINFO" name_extended="" sci_id="">
    <checkitem desc="RFC gateway hardening (secinfo content)" id="1.80.02d" operator="">
      <compliant>line like '%' and ( (text like_regexpr 'USER-HOST=[^*]' and text like_regexpr '[^-]HOST=[^*]') or text like_regexpr 'HOST=[*][.]')</compliant>
      <noncompliant>line like '%' and not ( (text like_regexpr 'USER-HOST=[^*]' and text like_regexpr '[^-]HOST=[^*]') or text like_regexpr 'HOST=[*][.]')</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for ms acl info -->
  <configstore name="MS_SECINFO" name_extended="" sci_id="">
    <checkitem desc="Message server hardening (acl_info content)" id="1.80.03d" operator="EXIST">
      <compliant>line like '%' and text like_regexpr 'HOST=[^*]'</compliant>
      <noncompliant>line like '%' and text = 'HOST=*'</noncompliant>
    </checkitem>
  </configstore>
  <!-- Checks for table authorization groups -->
  <configstore name="TDDAT_TABLES" name_extended="" sci_id="">
    <checkitem desc="Secure cryptographic keys" id="1.80.04a" not_found="positive" operator="">
      <compliant>TABNAME = 'SSF_PSE_D'  and CCLASS = 'SPSE' or CCLASS = 'YUSR' </compliant>
      <noncompliant>TABNAME = 'SSF_PSE_D'  and not ( CCLASS = 'SPSE' or CCLASS = 'YUSR' ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USR02)" id="1.20.03a" not_found="positive" operator="">
      <compliant>TABNAME = 'USR02'  and CCLASS = 'SPWD' or CCLASS = 'YUSR'</compliant>
      <noncompliant>TABNAME = 'USR02'  and not ( CCLASS = 'SPWD' or CCLASS = 'YUSR' ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USH02)" id="1.20.03b" not_found="positive" operator="">
      <compliant>TABNAME = 'USH02'  and CCLASS = 'SPWD' OR CCLASS = 'YUSR' </compliant>
      <noncompliant>TABNAME = 'USH02'  and not ( CCLASS = 'SPWD' or CCLASS = 'YUSR' ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (VUSER001)" id="1.20.03c" not_found="positive" operator="">
      <compliant>TABNAME = 'VUSER001'  and CCLASS = 'SPWD' or CCLASS = 'YUSR'</compliant>
      <noncompliant>TABNAME = 'VUSER001'  and not ( CCLASS = 'SPWD' or CCLASS = 'YUSR' ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USH02_ARC_TMP)" id="1.20.03d" not_found="positive" operator="">
      <compliant>TABNAME = 'USH02_ARC_TMP'  and CCLASS = 'SPWD' or CCLASS = 'YUSR'</compliant>
      <noncompliant>TABNAME = 'USH02_ARC_TMP'  and not ( CCLASS = 'SPWD' or CCLASS = 'YUSR' ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (USRPWDHISTORY)" id="1.20.03e" not_found="positive" operator="">
      <compliant>TABNAME = 'USRPWDHISTORY'  and CCLASS = 'SPWD' or CCLASS = 'YUSR'</compliant>
      <noncompliant>TABNAME = 'USRPWDHISTORY'  and not ( CCLASS = 'SPWD' or CCLASS = 'YUSR' ) </noncompliant>
    </checkitem>
    <checkitem desc="Restrict access to password hashes (VUSR02_PWD)" id="1.20.03f" not_found="positive" operator="">
      <compliant>TABNAME = 'VUSR02_PWD'  and CCLASS = 'SPWD' or CCLASS = 'YUSR' </compliant>
      <noncompliant>TABNAME = 'VUSR02_PWD'  and not ( CCLASS = 'SPWD' or CCLASS = 'YUSR' ) </noncompliant>
    </checkitem>
  </configstore>
</targetsystem>